### PR TITLE
Refactor: Moved a number of edx-platform imports to figures.compat

### PR DESCRIPTION
**Pull request by johnbaldwin**
Improved abstraction: This commit refactors how Figures imports edx-platform classes. It centralizes imports to figures.compat. The rest of Figures then imports the platform classes from figures.compat.

This is preliminary work to help abstract platform dependencies for Open edX named release structures and improve future testing efforts

Also bumped Figures release version to 0.4.dev7

### CODE REVIEW

File: figures/compat.py [hunk 1]
```diff
@@ -1,7 +1,9 @@
 """Figures compatability module
 
-This module serves to provide a common access point to functionality that
-differs from  different named Open edX releases
+This module serves to provide a common access point to edx-platform objects.
```
Comment (OmarIthawi) [inline figures/compat.py:3]: **Nit:** The module can now be called `openedx` instead of `compat` to reflect the newer name imho.

File: figures/models.py [hunk 2]
```diff
@@ -116,6 +116,8 @@ class SiteDailyMetrics(TimeStampedModel):
     # Should change this to default value of 0
     mau = models.IntegerField(blank=True, null=True)
 
+    # TODO: Add field for number of CDMs reported
```
Comment (OmarIthawi) [inline figures/models.py:119]: **Nit:** This doesn't seem to be related to the pull request overall. Was it added by mistake?

File: figures/sites.py [hunk 3]
```diff
@@ -241,3 +241,25 @@ def student_modules_for_course_enrollment(ce):
     Relies on the fact that course_ids are globally unique
     """
     return StudentModule.objects.filter(student=ce.user, course_id=ce.course_id)
+
+
+def site_certificates(site):
```
Comment (OmarIthawi) [inline figures/sites.py:246]: **Nit:** This function doesn't seem to be both unused and unrelated to the pull request. Was it added by mistake?

File: tests/test_sites.py [hunk 4]
```diff
@@ -31,10 +31,11 @@
 
 import organizations
 
-from openedx.core.djangoapps.content.course_overviews.models import (
-    CourseOverview,
-)
+# from openedx.core.djangoapps.content.course_overviews.models import (
+#     CourseOverview,
+# )
```
Comment (OmarIthawi) [inline tests/test_sites.py:36]: **Nit:** Probably not needed.

```suggestion
```

**Review by OmarIthawi — APPROVED** [review]
LGTM. Thanks John!

File: tests/test_sites.py (hunk 4 above)
Comment (johnbaldwin) [reply tests/test_sites.py:36]: I removed this dead code. Thanks!

File: figures/sites.py (hunk 3 above)
Comment (johnbaldwin) [reply figures/sites.py:246]: Not added by mistake, just a little extra that I remembered I wanted in there as I was refactoring

File: figures/models.py (hunk 2 above)
Comment (johnbaldwin) [reply figures/models.py:119]: Nope, on purpose, a reminder while I was in the code

File: figures/compat.py (hunk 1 above)
Comment (johnbaldwin) [reply figures/compat.py:3]: Good idea, I'll do that during another refactor
